### PR TITLE
feat(pubsub): add stub for the schema service

### PR DESCRIPTION
**Pull request by coryan**
This is the first step towards a `pubsub::SchemaClient`.

Part of the work for #5706

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5888)

### CODE REVIEW

File: google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+using ::google::cloud::testing_util::ScopedEnvironment;
+using ::google::cloud::testing_util::StatusIs;
+using ::testing::Not;
+
+bool UsingEmulator() {
+  return google::cloud::internal::GetEnv("PUBSUB_EMULATOR_HOST").has_value();
+}
+
+// TODO(#5706) - use the user-facing SchemaClient class. For now, as we
+//      bootstrap the classes just use the stub.
```
Comment (devbww) [inline google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc:37]: Clarify with a little rewording?

File: google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  auto const schema_id = pubsub_testing::RandomSchemaId(generator);
+  {
+    grpc::ClientContext context;
+    google::pubsub::v1::CreateSchemaRequest request;
+    request.set_parent("projects/" + project_id);
+    request.set_schema_id(schema_id);
+    request.mutable_schema()->set_type(google::pubsub::v1::Schema::AVRO);
+    request.mutable_schema()->set_definition(kTestAvroSchema);
+    auto schema = schema_admin->CreateSchema(context, request);
+    ASSERT_THAT(schema, StatusIs(StatusCode::kOk));
```
Comment (devbww) [inline google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc:68]: Nit:  Use `IsOk()` matcher?

**Review by devbww — APPROVED** [review]

File: google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc:37]: Perhaps it is just s/classes/classes, /.

Comment (coryan) [reply google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc]: Yup, that was it.

File: google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/pubsub/integration_tests/schema_admin_integration_test.cc]: Done, good idea.

**Pull request merged**